### PR TITLE
Use pkg-config to find dependencies

### PR DESCRIPTION
**Pull request by Youw**
- completely rework `setup.py` to use pkg-config for dependencies and build options (similar way how HIDAPI does it) when building HIDAPI from sources;
- use pkg-config when searching for HIDAPI package with `--use-system-hidapi`;
- switch to `hidraw`(-only) backend  as a default on Linux;

Fixes: #54
Closes: #55
Closes: #131
Fixes: #133

### CODE REVIEW

**Comment by Youw** [comment]
I've only properly tested this on Linux.
Will check it on Windows/macOS later.
Don't have any BSD to check it for now.

**Comment by Youw** [comment]
Just noticed #131 - I'll include that change in this PR as well as otherwise if would make a huge conflict in the code.

File: chid.pxd
Comment (Youw) [inline chid.pxd:7]: otherwise it yields warnings/errors since doesn't exactly match HIDAPI code

File: setup.py
Comment (Youw) [inline setup.py:62]: removed `-framework` commands from `CFLAGS` (moved to linker flags) as otherwise I get warnings like:
```sh
clang: warning: -framework IOKit: 'linker' input unused [-Wunused-command-line-argument]
```

File: setup.py
Comment (Youw) [inline setup.py:43]: Switched from using environment variables to `extra_compile_args`/`extra_link_args`.
Couldn't find the exact reason why env variables should used, except the fact that it was introduced 12 years ago during the initial commit.

**Comment by Youw** [comment]
> Just noticed https://github.com/trezor/cython-hidapi/pull/131 - I'll include that change in this PR as well as otherwise if would make a huge conflict in the code.

After going over #131 carefully, and taking into account that HIDAPI/libusb backend has a lot (even though very specific uses) and probably going to be available on Windows and macOS, I'm opting into keeping the behavior of `--with-libusb` as a complete fallback to "legacy" behavior (when both `hid` package is available for libusb and `hidraw` package for hidraw backend of HIDAPI).

As for future plans/default behavior - I'd consider explicitly providing a separate package (e.g. `hid.libusb`) on platforms that support it _in addition_ to `hid` package, which should be considered as best/default backend for those who don't care explicitly.

**Comment by Youw** [comment]
Still haven't checked BSD-like systems, but I'm confident this is **_ready to be checked/reviewed_**.

NOTE: there is no special handling of BSD-like  on HIDAPI/CMake build scripts (except NETBSD - recently there is a separate backend, but that's a separate story), so probably shouldn't be anything specific in `cython-hidapi` either.

File: README.rst
Comment (prusnak) [inline README.rst:54]:
```suggestion
   On Linux, the hidraw API is now used by default, and the libusb API is not
```

File: pyproject.toml
Comment (prusnak) [inline pyproject.toml:2]: This is not needed anymore, right?

```suggestion
requires = ["setuptools", "Cython"]
```

File: setup.py
Comment (prusnak) [inline setup.py:45]: Let's use f-strings:

```suggestion
        cmd = f"{pkg_config_exe} --exists '{package}'"
```

File: setup.py
Comment (prusnak) [inline setup.py:52]: Let's use f-strings:


```suggestion
        cmd = "{pkg_config_exe} {' '.join(options)} '{package}'"
```

File: setup.py
Comment (prusnak) [inline setup.py:79]:
```suggestion
        print("Without libusb is already a default, '--without-libusb' option is redundant and deprecated")
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by Youw** [comment]
BTW: I also had to include (partially unrelated) fix, to be able to run it locally.
It appears it fixes the "known" issue (https://github.com/trezor/cython-hidapi/actions/runs/6552643885/job/17796388898) and would unblock #160 as well.
Surely can be contributed to master as a separate commit if requested by code owner.

File: pyproject.toml
Comment (Youw) [reply pyproject.toml:2]: right, a leftover from an intermediate version

**Comment by Youw** [comment]
One more thing: if there're any code-style/indentation guids I should follow or a a tool to apply formatting - please do let me know.
Python is not my primary language, I've formatted the code based on the existing code and my personal preferences.

**Comment by prusnak** [comment]
I merged all your work into the `rework` branch. Thank you for all your effort.

I will try to build the wheels via your new method and if everything works just fine I'll merge this branch into master.

Thanks again!

**Comment by prusnak** [comment]
The CI successfully built the `rework` branch here: https://github.com/trezor/cython-hidapi/actions/runs/6842485940

So I merged `rework` into `master`. Thanks again!

**Comment by bearsh** [comment]
As this PR changes the default backend on linux, wouldn't it make sense to adjust the wheels action to pass `--with-libusb` to be backwards compatible? otherwise it would make sense to raise the version to 1.x as this is kind of a breaking change as e.g. the users udev rules may not work anymore.

**Comment by prusnak** [comment]
> As this PR changes the default backend on linux, wouldn't it make sense to adjust the wheels action to pass `--with-libusb` to be backwards compatible? otherwise it would make sense to raise the version to 1.x as this is kind of a breaking change as e.g. the users udev rules may not work anymore.

I want to keep the version number the same as hidapi version, so no bump to 1.x

@Youw what do you think about the broken udev rules? Should we keep using libusb as the default backend on Linux?

**Comment by prusnak** [comment]
> wouldn't it make sense to adjust the wheels action to pass `--with-libusb` to be backwards compatible?

Done in  https://github.com/trezor/cython-hidapi/commit/690824080c360c60276e4da0f873bd3df223a502

**Pull request merged**
